spec/rspec/matchers/built_in/change_spec.rb

```diff
@@ -356,6 +356,22 @@ def self.to_s
       expect { @instance.some_value = 6 }.not_to change(@instance, :some_value)
     end.to fail_with("expected `SomethingExpected#some_value` not to have changed, but did change from 5 to 6")
   end
+
+  it 'tolerates objects that do not change themselves' do
+    a = []
+    b = 1
+    expect { a << b }
+      .to change { b }
+      .to satisfy { |value| a.include?(value) }
+  end
```

Comment:
IMO this example specifies a bug.  The expectation here is that the block changed `b` in a way that satisfies the passed matcher.  It only makes sense for this expectation to pass if `b` did in fact change. If `b` did not change, then `expect { ... }.to change { b }` should not pass no matter what further customizations are chained on the matcher.  In this example, `b` did not change (and in fact, cannot change: numeric values can never be mutated, and `b` was not reassigned to a different value).

Reply:
Welcome back!

Agree. This gets especially confusing if the matcher would pass with a specifier, but would fail without one.

We actually made a change to address this very issue in a different way in #1132, but it seems it's not released as part of the latest release.